File: android/app/src/main/java/app/chompass/models/GoalFormulaReference.kt
package app.chompass.models

/**
 * Canonical formula text fragments for AI goal prompts — must stay aligned with deterministic code.
 * Unit-tested for parity with [ActivityLevel] and [NutritionConstants].
 */
object GoalFormulaReference {
    fun activityMultipliersLine(): String =
        ActivityLevel.entries.joinToString(", ") { level ->
            "${level.name.lowercase()} ${formatMultiplier(level.multiplier)}"
        }

    fun proteinPerKgLine(): String =
        ActivityLevel.entries.joinToString(", ") { level ->
            "${level.name.lowercase()} ${formatMultiplier(level.proteinPerKg)}"
        }

    fun calorieAdjustmentLine(): String {
        val kcalPerKg = NutritionConstants.KCAL_PER_KG_BODY_MASS.toInt()
        return "lose: -(weeklyChangeKg*$kcalPerKg/7); gain: +(weeklyChangeKg*$kcalPerKg/7)"
    }

    fun moderateActivityMultiplierRationale(): String =
        "Moderate uses ${formatMultiplier(ActivityLevel.MODERATE.multiplier)} (between FAO/WHO light 1.375 " +
            "and moderate 1.55) for desk-active users who exercise a few times per week."

    fun calorieSafetyLine(): String =
        "Never set calories below this user's BMR or ${CalorieSafety.ABSOLUTE_FLOOR_KCAL}. " +
            "If the weekly pace would break that floor, shrink the deficit instead of lowering the floor. " +
            "800 kcal is a medically supervised VLCD, not an app target."

    private fun formatMultiplier(value: Double): String =
        if (value % 1.0 == 0.0) value.toInt().toString() else value.toString()
}

File: app/src/lib/chompass-core/goal-formula-reference.js
// @ts-check
/**
 * Canonical formula text fragments for AI goal prompts — mirrors Android
 * GoalFormulaReference.kt. Must stay aligned with formulas.js constants and
 * testdata/parity/goal-formula-prompt-fragments.json.
 */
import { PAL_MULTIPLIERS, PROTEIN_G_PER_KG, KCAL_PER_KG_BODY_MASS, CALORIE_ABSOLUTE_FLOOR_KCAL } from "./formulas.js";

/** @param {number} value */
function formatMultiplier(value) {
  return value % 1 === 0 ? String(value) : String(value);
}

export function activityMultipliersLine() {
  return Object.entries(PAL_MULTIPLIERS)
    .map(([k, v]) => `${k} ${formatMultiplier(v)}`)
    .join(", ");
}

export function proteinPerKgLine() {
  return Object.entries(PROTEIN_G_PER_KG)
    .map(([k, v]) => `${k} ${formatMultiplier(v)}`)
    .join(", ");
}

export function calorieAdjustmentLine() {
  const kcalPerKg = Math.trunc(KCAL_PER_KG_BODY_MASS);
  return `lose: -(weeklyChangeKg*${kcalPerKg}/7); gain: +(weeklyChangeKg*${kcalPerKg}/7)`;
}

export function moderateActivityMultiplierRationale() {
  const moderate = formatMultiplier(PAL_MULTIPLIERS.moderate);
  return (
    `Moderate uses ${moderate} (between FAO/WHO light 1.375 ` +
    `and moderate 1.55) for desk-active users who exercise a few times per week.`
  );
}

export function calorieSafetyLine() {
  return (
    `Never set calories below this user's BMR or ${CALORIE_ABSOLUTE_FLOOR_KCAL}. ` +
    `If the weekly pace would break that floor, shrink the deficit instead of lowering the floor. ` +
    `800 kcal is a medically supervised VLCD, not an app target.`
  );
}
